### PR TITLE
optimize the logic of main.py

### PR DESCRIPTION
**Pull request by Randsome-Arisa**
At the line 218 of `main.py`, building optimizer and lr scheduler before `if args.eval:` , but if the mode is eval, the `load_checkpoint()` don't need the parameters optimizer and scheduler.
If there are only `split='val'` in `refs.p` but no `split='train'`, the original logic will be wrong because of the setting of scheduler:
`scheduler = torch.optim.lr_scheduler.LambdaLR(optimizer, lambda step: (1 - step / (len(train_loader) * args.epoch)) ** 0.9)`
the `len(train_loader)` will be 0, and there is bug that devided by 0.
So I move the setting of optimizer and lr scheduler after the `if args.eval:` to fix the bug.

### CODE REVIEW

**Comment by 1233211234569** [comment]
除以0，会导致iou为0吗
